crates/evm/evm/src/executors/invariant/error.rs

```diff
@@ -214,12 +214,18 @@ impl InvariantFuzzError {
                 .call_raw_committing(*sender, *addr, bytes.clone(), U256::ZERO)
                 .expect("bad call to evm");
 
-            // Checks the invariant. If we exit before the last call, all the better.
+            // Checks the invariant. If we revert or fail before the last call, all the better.
             if let Some(func) = &self.func {
-                let error_call_result = executor
+                let call_result = executor
                     .call_raw(CALLER, self.addr, func.clone(), U256::ZERO)
                     .expect("bad call to evm");
-                if error_call_result.reverted {
+                let is_success = executor.is_raw_call_success(
+                    self.addr,
+                    call_result.state_changeset.clone().unwrap(),
```

Comment:
do we need the clone here?

Reply:
the next param is `&call_result` which borrows the value so  if not cloning fails with
```
225 |                     &call_result,
    |                     ^^^^^^^^^^^^ value borrowed here after partial move
```

Reply:
hmm, the `pub fn is_raw_call_success(` fn is not very good, I see why we need the state as a separate arg

but this means for this call, we can use `mut call_result.state_changeset.take().unwrap()`

Reply:
thank you, this makes sense, I made following change in https://github.com/foundry-rs/foundry/pull/7257/commits/9d2b073c4a3d9afb4fd53464b76b5bd28a179177 + made test more robust as they can change in less steps